### PR TITLE
Develop daq monitoring

### PR DESCRIPTION
**Pull request by mexanick**
## Description

A working prototype of daq monitoring, ready for initial review. As it compiles and runs fine, eventually can be merged in order to prevent huge PRs. 
Requires:
```bash
xhal.x86_64                                                            3.2.2-1.0.88.dev.a08d8adgit.centos7.gcc4_8_5                                                       @gemos-test
xhal-devel.x86_64                                                      3.2.2-1.0.88.dev.a08d8adgit.centos7.gcc4_8_5                                                       @gemos-test
``` 
currently installed at `gem904daq01`

`gem::daqmon::DaqMonitor` class provides the monitoring functions for each AMC board in the uTCA crate (shelf). The monitoring capabilities currently reduced to only few DAQ registers, but will be extended in nearby future. `ShelfMonitor` application provides interface to create and run the `DaqMonitor` for all AMCs in the crate (shelf) and provides a web-interface for visualization. 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

Tested by running XDAQ application at `gem904daq01`.
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jsturdy** [comment]
@mexanick, can you rebase this locally on top of the current `cms-gem-daq-project/develop` branch?

File: gempython/tests/getVFATMask.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/env python
```
Comment (jsturdy) [inline gempython/tests/getVFATMask.py]: This (file) shouldn't be here, the PR introducing it was explicitly done against a specific, short-term branch

File: gempython/tools/amc_user_functions_xhal.py [hunk 2]
```diff
@@ -11,6 +11,39 @@
 gMAX_RETRIES = 5
```
Comment (jsturdy) [inline gempython/tools/amc_user_functions_xhal.py]: Likewise this...

File: gempython/tools/optohybrid_user_functions_xhal.py [hunk 3]
```diff
@@ -1,8 +1,7 @@
-import os, sys
-
 from gempython.tools.amc_user_functions_xhal import *
```
Comment (jsturdy) [inline gempython/tools/optohybrid_user_functions_xhal.py]: This too...

File: gempython/tools/vfat_user_functions_xhal.py [hunk 4]
```diff
@@ -16,7 +16,16 @@ def __init__(self, cardName, link, debug=False):
```
Comment (jsturdy) [inline gempython/tools/vfat_user_functions_xhal.py]: Pretty much anything not in `gemdaqmonitor` is going to get my rejection

**Comment by mexanick** [comment]
ok, I think it is clean now. Looks like no other changes than new files added by my are in

File: gemdaqmonitor/src/common/ShelfMonitor.cc [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  DEBUG("gem::daqmon::ShelfMonitor : Destructor called");
+  // make sure to empty the v_supervisedApps  vector and free the pointers
+}
+
+
+// This is the callback used for handling xdata:Event objects
+void gem::daqmon::ShelfMonitor::actionPerformed(xdata::Event& event)
+{
+  if (event.type() == "setDefaultValues" || event.type() == "urn:xdaq-event:setDefaultValues") {
+    DEBUG("GEMSupervisor::actionPerformed() setDefaultValues" <<
```
Comment (jsturdy) [inline gemdaqmonitor/src/common/ShelfMonitor.cc]: Some residual copy/paste leftovers...

**Pull request merged**